src/lib/litegraph/src/contextMenuCompat.ts

```diff
@@ -0,0 +1,141 @@
+import type { IContextMenuValue } from './interfaces'
+
+/**
+ * Simple compatibility layer for legacy getCanvasMenuOptions and getNodeMenuOptions monkey patches.
+ * To disable legacy support, set ENABLE_LEGACY_SUPPORT = false
+ */
+const ENABLE_LEGACY_SUPPORT = true
+
+type AnyFunction = (...args: any[]) => any
+
+class LegacyMenuCompat {
+  private originalMethods = new Map<string, AnyFunction>()
+  private hasWarned = new Set<string>()
+  private currentExtension: string | null = null
+  private isExtracting = false
```

Comment:
I don't know that `isExtracting` is clear, descriptive, and unambiguous in context. But also, I think there's a way to do this without needing to have this semaphore.